Automattic/jetpack
# Socia: Refactor notice dismiss functionality to support time periods

This change updates the current functionality for the notice dismiss feature, so it can be given a timestamp to send it away for some time rather than forever.

## Proposed changes:

* Added a new parameter to the endpoint that is the reappearance time

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Open up your local site with this PR. If you already dismissed the Instagram notice run `jetpack docker wp option delete jetpack_social_dismissed_notices`
* Open the Admin page or the editor, and dismiss the notice.
  * Run `jetpack docker wp option get jetpack_social_dismissed_notices`, it should return an object with instagram in it, and the value should be 0.
  * Refresh the page, the notice should not come back.
* Clear the option again with the command provided above.
* In `projects/js-packages/publicize-components/src/components/form/index.js` change `dismissNotice( 'instagram' )` to `dismissNotice( 'instagram', 1 )`.
  * This should dismiss the notice for 1 minute. Run the command above to check the actual value in the option, it should be the Unix timestamp of the current time + one minute.
  * Refresh the page before the minute elapsed, the notice should not appear.
  * Refresh after 1 minute, the notice should come back again. Clicking close now should send it away for another minute
